### PR TITLE
commit

### PR DESCRIPTION
**Pull request by MaliukDaria**

### CODE REVIEW

**Comment by MaliukDaria** [comment]
Travis не проходит из-за проблем с подключением JSTL библиотеки. Maven clean Package на Intellij проходит без проблем

File: src/main/java/com/online/store/controllers/product/controllers/AddProductController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.online.store");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:15]:
```suggestion
    private ProductService productService =
            (ProductService) injector.getInstance(ProductService.class);
```

Comment (dariiabutorina) [reply src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:15]: Let's remove these modifiers for all of the service objects

File: src/main/java/com/online/store/controllers/product/controllers/AddProductController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String name = req.getParameter("name");
+        double price = Double.parseDouble(req.getParameter("price"));
```
Comment (dariiabutorina) [inline src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:27]: Let's create a check for the input data
We can't work with the empty Strings

File: src/main/java/com/online/store/controllers/user/controllers/RegisterUserController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/users/registration.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        String repeatPassword = req.getParameter("repeatPassword");
```
Comment (dariiabutorina) [inline src/main/java/com/online/store/controllers/user/controllers/RegisterUserController.java:31]: Same check

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Not bad overall :)

File: src/main/java/com/online/store/controllers/product/controllers/AddProductController.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:15]: you can leave them

File: src/main/java/com/online/store/controllers/product/controllers/AddProductController.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:27]: this should be done by jsp 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

**Comment by RomanDubovskyi** [comment]
travis failed

File: src/main/java/com/online/store/controllers/shoppingcart/controller/AddToShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+public class AddToShoppingCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.online.store");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+    private ShoppingCart shoppingCart;
```
Comment (RomanDubovskyi) [inline src/main/java/com/online/store/controllers/shoppingcart/controller/AddToShoppingCartController.java]: we can make this field local

File: src/main/java/com/online/store/controllers/user/controllers/DeleteUserController.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.online.store");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Long userId = Long.valueOf(req.getParameter("id"));
+        userService.delete(userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/online/store/controllers/user/controllers/DeleteUserController.java:21]: let's delete here shoppingCart as well

File: src/main/webapp/WEB-INF/web.xml [hunk 6]
```diff
@@ -7,9 +8,97 @@
...
+        <url-pattern> /products/delete</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>buyProduct</servlet-name>
+        <servlet-class>com.online.store.controllers.shoppingcart.controller.AddToShoppingCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>buyProduct</servlet-name>
+        <url-pattern>/products/buy</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/web.xml]:
```suggestion
        <url-pattern>/shoppingCart/products/add</url-pattern>
```

File: src/main/webapp/WEB-INF/web.xml [hunk 7]
```diff
@@ -7,9 +8,97 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProductFromSoppingCart</servlet-name>
+        <servlet-class>com.online.store.controllers.shoppingcart.controller.DeleteProductFromShoppingCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>deleteProductFromSoppingCart</servlet-name>
+        <url-pattern>/shoppingCart/delete</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/web.xml]:
```suggestion
        <url-pattern>/shoppingCart/products/delete</url-pattern>
```

**Comment by MaliukDaria** [comment]
> travis failed

Travis не проходит из-за проблем с подключением JSTL библиотеки. Maven clean Package на Intellij проходит без проблем. Я попробую исправить это переустановив Intellij

**Review by RomanDubovskyi — COMMENTED** [review]
travis failed

File: src/main/java/com/online/store/controllers/product/controllers/AddProductController.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:27]: done

File: src/main/java/com/online/store/controllers/user/controllers/RegisterUserController.java (hunk 3 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/user/controllers/RegisterUserController.java:31]: done

File: src/main/java/com/online/store/controllers/shoppingcart/controller/AddToShoppingCartController.java (hunk 4 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/shoppingcart/controller/AddToShoppingCartController.java]: done

File: src/main/java/com/online/store/controllers/user/controllers/DeleteUserController.java (hunk 5 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/user/controllers/DeleteUserController.java:21]: done

File: src/main/webapp/WEB-INF/web.xml (hunk 6 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/web.xml]: done

File: src/main/webapp/WEB-INF/web.xml (hunk 7 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/web.xml]: done

File: src/main/webapp/WEB-INF/web.xml (hunk 6 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/web.xml]: done

File: src/main/webapp/WEB-INF/web.xml (hunk 7 above)
Comment (MaliukDaria) [reply src/main/webapp/WEB-INF/web.xml]: done

File: src/main/java/com/online/store/controllers/user/controllers/DeleteUserController.java (hunk 5 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/user/controllers/DeleteUserController.java:21]: done

File: src/main/java/com/online/store/controllers/shoppingcart/controller/AddToShoppingCartController.java (hunk 4 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/shoppingcart/controller/AddToShoppingCartController.java]: done

File: src/main/java/com/online/store/controllers/user/controllers/RegisterUserController.java (hunk 3 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/user/controllers/RegisterUserController.java:31]: done)

File: src/main/java/com/online/store/controllers/product/controllers/AddProductController.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/controllers/product/controllers/AddProductController.java:27]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**